FestivalPlanner/screenings/views.py

```diff
@@ -306,23 +336,37 @@ class ScreeningDetailView(LoginRequiredMixin, SingleObjectMixin, FormView):
     http_method_names = ['get', 'post']
     fan_action = FanAction('update')
     update_by_attends = {True: 'joins', False: "couldn't come"}
+    update_by_has_ticket = {True: 'got a ticket', False: 'sold a ticket'}
+    update_by_confirmed = {True: 'had a ticket confirmed', False: 'had a ticket unconfirmed'}
     fans = None
     object = None
     screening = None
     initial_attendance_by_fan = None
+    initial_tickets_by_fan = None
+    initial_confirmed_by_fan = None
 
     def setup(self, request, *args, **kwargs):
         super().setup(request, *args, **kwargs)
-        initialize_log(request.session, 'Update attendances')
-        self.fans = get_present_fans(self.request.session)
+        initialize_log(request.session, 'Update attendance statuses')
+        session = self.request.session
```

Comment:
Please explain the order of these lines.

Reply:
No good reason.
Improved the code by using only the `session` of the `request` from the parameters.